lib/vrl/stdlib/src/reverse_dns.rs

```diff
@@ -0,0 +1,90 @@
+use dns_lookup::lookup_addr;
+use std::net::IpAddr;
+
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ReverseDns;
+
+impl Function for ReverseDns {
+    fn identifier(&self) -> &'static str {
+        "reverse_dns"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::BYTES,
+            required: true,
+        }]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[Example {
+            title: "Example",
+            source: r#"reverse_dns!("127.0.0.1")"#,
+            result: Ok("localhost"),
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+
+        Ok(Box::new(ReverseDnsFn { value }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ReverseDnsFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for ReverseDnsFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let ip: IpAddr = self
+            .value
+            .resolve(ctx)?
+            .try_bytes_utf8_lossy()?
+            .parse()
+            .map_err(|err| format!("unable to parse IP address: {}", err))?;
+
+        let host =
+            lookup_addr(&ip).map_err(|err| format!("unable to perform a lookup : {}", err))?;
+
+        Ok(host.into())
+    }
+
+    fn type_def(&self, state: &state::Compiler) -> TypeDef {
+        self.value
+            .type_def(state)
+            .fallible_unless(value::Kind::Bytes)
```

Comment:
Shouldn't this function _always_ be fallible?

Reply:
Doh, yeah, sorry, I should have went back over this before requesting review 😓